scrapy/utils/decorators.py

```diff
@@ -43,3 +43,31 @@ def inthread(func):
     def wrapped(*a, **kw):
         return threads.deferToThread(func, *a, **kw)
     return wrapped
+
+
+try:
+    from dataclasses import fields as dataclass_fields
+except ImportError:
+    pass
+else:
+    def subscriptable_dataclass(cls):
+        """
+        Allow dictionary-like access on dataclass instances
+        """
+
+        def __getitem__(self, key):
+            field_names = [f.name for f in dataclass_fields(self)]
```

Comment:
`if key in [f.name for f in dataclass_fields(self)]` is repeated at least 5 times in the Scrapy codebase, I wonder if we should make a small helper for that.